sentry/src/main/java/io/sentry/SentryEnvelopeItem.java

```diff
@@ -151,6 +153,50 @@ public static SentryEnvelopeItem fromUserFeedback(
     return new SentryEnvelopeItem(itemHeader, () -> cachedItem.getBytes());
   }
 
+  public static SentryEnvelopeItem fromAttachment(
+      @NotNull ILogger logger, final @NotNull Attachment attachment) {
+
+    final CachedItem cachedItem =
+        new CachedItem(
+            () -> {
+              if (attachment.getBytes() != null) {
+                return attachment.getBytes();
+              } else if (attachment.getPathname() != null) {
+                try (FileInputStream fileInputStream =
+                        new FileInputStream(attachment.getPathname());
+                    BufferedInputStream inputStream = new BufferedInputStream(fileInputStream);
+                    ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+
+                  byte[] bytes = new byte[1024];
+                  int length;
+                  int offset = 0;
+                  while ((length = inputStream.read(bytes)) != -1) {
+                    outputStream.write(bytes, offset, length);
+                  }
+                  return outputStream.toByteArray();
+                } catch (IOException | SecurityException exception) {
```

Comment:
Could this throw OOM here too? Given that we'll have to allocate a byte array for the whole size of this file?

Reply:
Yes, it could. If you try to send a 10GB file. Anyway relay has a default 50Mib limit for envelopes, see https://docs.sentry.io/product/relay/options/. So we could anyway discard files that are bigger than that. 
I would rather avoid catching `OutOfMemoryError` and fix the underlying issue. We allocate memory in 1024 byte blocks inside a loop, if we run out of memory in there the whole application is going to be in trouble. If we would allocate all the memory for the file in one big chunk and we would get an `OutOfMemoryError` we could be fine, but we don't do that.